utils/featureFlags.ts

```diff
@@ -0,0 +1,33 @@
+import { right } from "fp-ts/lib/Either";
+import { fromEither, TaskEither } from "fp-ts/lib/TaskEither";
+import { InstallationId } from "../generated/notifications/InstallationId";
+import { NHPartitionFeatureFlag } from "./config";
+
+/**
+ *
+ * @param enabledFeatureFlag The feature flag currntly enabled
+ * @param sha the installation id of the user
+ * @returns `true` if the user is enabled for the new feature, `false` otherwise
+ */
+export const getIsInActiveSubset = (
+  isUserATestUser: (InstallationId) => TaskEither<Error, boolean>
+) => (
+  enabledFeatureFlag: NHPartitionFeatureFlag,
+  sha: InstallationId
+): TaskEither<Error, boolean> => {
+  switch (enabledFeatureFlag) {
+    case NHPartitionFeatureFlag.all:
+      return fromEither(right(true));
+
+    case NHPartitionFeatureFlag.beta:
+      return isUserATestUser(sha);
+
+    case NHPartitionFeatureFlag.canary:
+      // Todo
+      return fromEither(right(false));
```

Comment:
```suggestion
      return taskEither.of(false);
```

Reply:
Also: remember that a beta user is also a canary user, so we may want to do something like
```ts
return canaryRegex.test(sha)
  ? taskEither.of(true)
  : isUserATestUser(sha)
```

Reply:
Yes, you're absolutely right, we need to maintain beta users also when we switch to canary feature flag!